### PR TITLE
feat: storage-inspect with support for check

### PR DESCRIPTION
**Pull request by odyslam**
## Motivation

Ensure that we don't change the storage layout by mistake

## Solution

If a PR changes the storage layout of a smart contract, the check will fail.  The developer will need to run `yarn storage-inspect:generate` to create a new storage-layout file and commit that, so the test can pass. This forces the team to explicitly acknowledge the fact that the storage layout changed.

## PR Checklist

- [x] Added Tests // Tested Locally
- [ ] Updated Documentation
- [ ] Updated CHANGELOG.md for the appropriate package

### CODE REVIEW

**Review by prestwich — CHANGES_REQUESTED** [review]
can you document the difference between snapshot and storage-inspect?

**Comment by odyslam** [comment]
Updated the READMEs for bridge, contract @prestwich

File: packages/contracts-bridge/README.md [hunk 1]
```diff
@@ -4,9 +4,41 @@ Solidity implementations of the [Nomad Bridge](https://app.nomad.xyz).
...
+
+- Run `yarn build:accumulator-cli` from the root directory of the monorepo. It will build a rust-based cli tool that creates Sparse Merkle Tree proofs for arbitrary data. It's used in our testing suite via the `--ffi` flag for Forge. The binary is built in there `/scripts` top-level directory of the monorepo
+- `--ffi` means that Forge will run arbitrary shell commands as part of the testing suite. You should never run `forge --ffi` without knowing what exactly are the shell commands that will be executed, as the testing suite could be malicious and execute malicious commands. This is why the feature is disabled by default and must be explicitly enabled.
+- `yarn test` will run all tests. Note that `--ffi` is enabled by default,
+- `yarn snapshot --check` will run the test suite and verify that it doesn't produce a different gas snapshot from the existing (`.gas-snapshot`)
+- `yarn snapshot` will create a new `.gas-snapshot`. You can inspect the different gas usage via `git diff`
+- 'yarn snapshot:check' will run the test suite and check gas consumption against the **existing** `.gas-snapshot`. It will `pass` only if there is no change in the gas consumption
```
Comment (prestwich) [inline packages/contracts-bridge/README.md:22]: duplicate of 2 lines up?

File: packages/contracts-bridge/README.md [hunk 2]
```diff
@@ -4,9 +4,40 @@ Solidity implementations of the [Nomad Bridge](https://app.nomad.xyz).
 
 ### Setup
 
-- See repo setup
-- `brew install jq` &nbsp; OR &nbsp; `sudo apt-get install jq`
```
[the remark this answers is not in the document]
Comment (odyslam) [reply packages/contracts-bridge/README.md:8]: Where do we use jq?

[the remark this answers is not in the document]
Comment (prestwich) [reply packages/contracts-bridge/README.md:8]: lol guess we don't anymore 🤷‍♀️

**Review by prestwich — APPROVED** [review]

**Pull request merged**